provider/source/src/units/data.rs

```diff
@@ -109,6 +109,8 @@ impl crate::IterableDataProviderCached<UnitsDisplayNameV1Marker> for SourceDataP
                         && !["per", "times", "power"]
                             .iter()
                             .any(|&prefix| key.starts_with(prefix))
+                            // Each unit should be prefixed with the category and a hyphen
+                            && key.contains("-")
```

Comment:
this shouldn't be required, as the `split_once` below already fails if there's no `-`

Reply:
I noticed this, actually, maybe we will not need this filter at all.

Reply:
done